src/Database/QueryCompiler.php

```diff
@@ -234,12 +256,17 @@ protected function _buildJoinPart($parts, $query, $generator)
                 $join['table'] = '(' . $join['table'] . ')';
             }
 
-            $joins .= sprintf(' %s JOIN %s %s', $join['type'], $join['table'], $join['alias']);
+            $joins .= sprintf(' %s JOIN (%s %s', $join['type'], $join['table'], $join['alias']);
```

Comment:
Could we only use parenthesis on join expressions that contain child associations? I think this change is why so many tests are failing.

Reply:
> Could we only use parenthesis on join expressions that contain child associations? I think this change is why so many tests are failing.

Yes. Te query join structure was changed from
```
LEFT JOIN teams Teams 
ON Teams.id = (Athletes.team_id) 

INNER JOIN juridical_people JuridicalPeople 
ON JuridicalPeople.id = (Teams.juridical_person_id)
```
to
```
LEFT JOIN (teams Teams 
	INNER JOIN (juridical_people JuridicalPeople) 
	ON JuridicalPeople.id = (Teams.juridical_person_id)
) 
ON Teams.id = (Athletes.team_id) 
```
and it's working fine to me.
Another way to do this, would be inserting a subquery, 
```
LEFT JOIN 
(
	SELECT 
		...
		...
	FROM teams Teams
	INNER JOIN juridical_people JuridicalPeople
	ON JuridicalPeople.id = Teams.juridical_person_id
) Teams  
ON Teams.id = (Athletes.team_id) 
```
but i think it would cause many performance troubles.